src/test/java/core/basesyntax/converter/FruitParserTest.java

```diff
@@ -0,0 +1,162 @@
+package core.basesyntax.converter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.converter.impl.FruitParserImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+class FruitParserTest {
+    private static final String BALANCE_OPERATION = "b";
+    private static final String PURCHASE_OPERATION = "p";
+    private static final String BANANA_FRUIT_NAME = "banana";
+    private static final int BANANA_FRUIT_QUANTITY = 20;
+    private static final String APPLE_FRUIT_NAME = "apple";
+    private static final int APPLE_FRUIT_QUANTITY = 13;
+    private static final String CSV_HEADER = "type,fruit,quantity";
+    private static final String BALANCE_BANANA_LINE = String.format("%s,%s,%s",
+            BALANCE_OPERATION, BANANA_FRUIT_NAME, BANANA_FRUIT_QUANTITY);
+    private static final String PURCHASE_APPLE_LINE = String.format("%s,%s,%s",
+            PURCHASE_OPERATION, APPLE_FRUIT_NAME, APPLE_FRUIT_QUANTITY);
+    private static FruitParser fruitParser;
+
+    @BeforeAll
+    public static void initParser() {
+        fruitParser = new FruitParserImpl();
+    }
+
+    @Test
+    public void parseFruitTransactions_nullValuePassed_notOk() {
+        Exception actualException = assertThrows(IllegalArgumentException.class,
+                () -> fruitParser.parseFruitTransactions(null));
+        assertEquals("Passed method argument is null", actualException.getMessage());
+    }
+
+    @Test
+    public void parseFruitTransactions_nullValueInList_ok() {
+        List<String> linesList = new ArrayList<>(Arrays.asList(
+                CSV_HEADER,
+                BALANCE_BANANA_LINE,
+                PURCHASE_APPLE_LINE));
+        List<FruitTransaction> expectedList = new ArrayList<>();
+        expectedList.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                BANANA_FRUIT_NAME, BANANA_FRUIT_QUANTITY));
+        expectedList.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                APPLE_FRUIT_NAME, APPLE_FRUIT_QUANTITY));
+        List<FruitTransaction> actualList = fruitParser.parseFruitTransactions(linesList);
+        assertEquals(expectedList, actualList);
+    }
+
+    @Test
+    public void parseFruitTransactions_noCsvHeader_ok() {
+        List<String> linesList =
+                new ArrayList<>(Arrays.asList(BALANCE_BANANA_LINE, PURCHASE_APPLE_LINE));
+
+        List<FruitTransaction> expectedList =
+                new ArrayList<>(Arrays.asList(
```

Comment:
why do you need `new ArrayList<>` and do not use simple `Arrays.asList` or `List.of`

Reply:
I used it because some methods in my logic require non abstract lists